lib/compute/index.js

```diff
@@ -0,0 +1,793 @@
+      }
+    });
+};
+
+/**
+ * Get a list of addresses.
+ *
+ * @throws {Error} if a malformed filter is provided.
+ *
+ * @param {object} options - Address search options.
+ * @param {number=} options.maxResults - Maximum number of addresses to return.
+ * @param {object=} options.filter - Search filter.
+ * @param {string} options.filter.fieldName - Address field to consider in this
+ *     filter.
+ * @param {string} options.filter.operatorString - Filter operator, can be
+ *     either 'eq' or 'ne'.
+ * @param {string} options.filter.literalString - String to compare the address
+ *     field to. Can be a regular expression.
+ * @param {string=} pageToken - Page identifier used in paginated search.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * var callback = function(err, addresses) {
+ *   // `addresses` is an array of `Address` objects.
+ * };
+ *
+ * var compute = gcloud.compute({
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * compute.getAddresses(
+ *   {
+ *     filter: {
+ *       fieldName : 'name',
+ *       operatorString : 'eq',
+ *       literalString : 'address-[0-9]'
+ *     }],
+ *   }, callback);
+ */
+Compute.prototype.getAddresses = function(options, callback) {
+  if (!callback) {
+    callback = options;
+  }
+
+  var query = {};
+
+  if (options) {
+    if (util.is(options.maxResults, 'number')) {
+      query.maxResults = options.maxResults;
+    }
+    if (options.filter) {
+      if (!util.is(options.filter.fieldName, 'string') ||
+          !util.is(options.filter.operatorString, 'string') ||
+          !util.is(options.filter.literalString, 'string')) {
+        throw new Error(
+          'A filter must have name, comparisonString and literal fields.');
+      }
+      if (options.filter.operatorString !== 'eq' &&
+        options.filter.operatorString !== 'ne') {
+        throw new Error('Filter operator must be either eq or ne');
+      }
+      query.filter =
+        options.filter.fieldName + ' ' +
+        options.filter.operatorString + ' \'' +
+        options.filter.literalString + '\'';
+    }
+    if (util.is(options.pageToken, 'string')) {
+      query.pageToken = options.pageToken;
+    }
+  }
+
+  var self = this;
+  this.makeReq_(
+    'GET',
+    '/aggregated/addresses',
+    query,
+    null, function(err, resp) {
+      if (err) {
+        callback(err);
+        return;
+      }
+      if (resp.items) {
+        var addresses = [];
+        for (var regionName in resp.items) {
+          var region = self.region(regionName);
+          var regionAddresses = resp.items[regionName].addresses;
+          if (regionAddresses) {
+            for (var i = 0; i < regionAddresses.length; i++) {
+              var metadata = regionAddresses[i];
+              var address = region.address(metadata.name);
+              address.metadata = metadata;
+              addresses.push(address);
+            }
+          }
+        }
+        callback(null, addresses);
```

Comment:
Filters are now simple strings, I added auto pagination support and removed filter validation so that we let the API catch those errors. 
I added no reference to the API docs yet. Were you thinking about adding a reference on a per-parameter basis or something like: *"For a detailed description of method options see [API REFERENCE]()."* on top of the docs that describe a method?